### PR TITLE
feat!: add evidence params to blocklist

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/2418 by adding evidence params to the blocklist.

### CODE REVIEW

**Review by cmwaters — DISMISSED** [review]
LGTM

File: x/paramfilter/test/param_filter_test.go
Comment (evan-forbes) [reply x/paramfilter/test/param_filter_test.go:85]: this is weird, as it seems like the evidence params are getting updated in endblock :thinking: 

https://github.com/celestiaorg/cosmos-sdk/blob/c525ad72ab1449d31115815f5832a68bc79d4ffd/baseapp/baseapp.go#L457-L488

File: x/paramfilter/test/param_filter_test.go
Comment (staheri14) [inline x/paramfilter/test/param_filter_test.go:113]: If it could be converted to an error type, it would enhance code readability and make potential future refactoring easier.

File: x/paramfilter/test/param_filter_test.go
Comment (staheri14) [inline x/paramfilter/test/param_filter_test.go:42]: [Question] Can you please elaborate how this test differs from the first test? Thanks

**Review by staheri14 — COMMENTED** [review]
LGTM! added some suggestion and question.
I'd defer it to other reviewers for final approval.

File: x/paramfilter/test/param_filter_test.go
Comment (rootulp) [reply x/paramfilter/test/param_filter_test.go:85]: In theory it should be possible to modify evidence params if I could encode the updated evidence params in the appropriate format (I think via the legacy Amino JSON encoder) but I couldn't get it to work with the permutations I tried in this test. Happy to convert to draft if you think it's important to have confidence that we could modify evidence params prior to adding the evidence params key to the blocklist.

Can convert this back to draft and try to figure out encoding / decoding if blocking.

File: x/paramfilter/test/param_filter_test.go
Comment (rootulp) [reply x/paramfilter/test/param_filter_test.go:42]: The first test uses the first blocked param from the list of blocked params.

The second test (this test) uses a different blocked param. Specifically the validator consensus params.

File: x/paramfilter/test/param_filter_test.go
Comment (rootulp) [reply x/paramfilter/test/param_filter_test.go:113]: hmm it's defined here: https://github.com/rootulp/celestia-app/blob/90b761bb7c2918e0ca75fc488801cd9b914431c5/x/paramfilter/types.go#L15

idk if it can be "converted to an error type" b/c 

```
// Register returns an error instance that should be used as the base for
// creating error instances during runtime.
```

but I can replace the error assertions with this diff if you prefer:

```diff
-			require.Contains(t, err.Error(), "parameter can not be modified")
+			require.ErrorIs(t, err, paramfilter.ErrBlockedParameter)
```

Comment (rootulp) [reply x/paramfilter/test/param_filter_test.go]: Attempted to address via https://github.com/celestiaorg/celestia-app/pull/2453/commits/4c452c5f4dc9de1225f91ea2618c6fca79297f44. Please unresolve if that doesn't address your concern.

File: x/paramfilter/test/param_filter_test.go
Comment (evan-forbes) [reply x/paramfilter/test/param_filter_test.go]: nah, we have confidence in the paramfilter as is, so just adding it should be sufficient

**Review by evan-forbes — APPROVED** [review]
before we merge imo we should either:

1) decide to backport this and attempt to get validators on the testnets to update
2) write an if statement to make this compatible with v2

I'm good with either, so I'm approving! will defer to the author and other reviewers. If we go with option 1 we should expidite the next release candidate and mocha-4 upgraded asap

**Comment by rootulp** [comment]
Agreed.

> write an if statement to make this compatible with v2

I'll attempt to do this now b/c I don't have confidence in my ability to coordinate all mocha-4 validators to upgrade to a new release and since mocha-4 is a live network, someone could submit a param change proposal with evidence params at any moment prior to all validators upgrading to a release with this PR. In that scenario, I expect a fork would occur.

**Comment by evan-forbes** [comment]
> I expect a fork would occur.

we'd have to hardfork to fix it yeah

File: x/paramfilter/test/param_filter_test.go
Comment (staheri14) [reply x/paramfilter/test/param_filter_test.go:42]: > The first test uses the first blocked param from the list of blocked params.

In the first test, it looks like it ranges through all of the blocked params, doesn't it? 
```go
                for _, p := range testApp.BlockedParams() {
			p := testProposal(proposal.NewParamChange(p[0], p[1], "value"))
			err := handler(ctx, p)
			require.Error(t, err)
			require.Contains(t, err.Error(), "parameter can not be modified")
		}
```

Comment (rootulp) [reply x/paramfilter/test/param_filter_test.go:42]: oh true! I can remove the second test case b/c it should be covered by test case one.

Comment (rootulp) [reply x/paramfilter/test/param_filter_test.go]: Resolved by f0a9951bd65a3e9ba6bbc265ab1bd3d255a5e212

File: x/paramfilter/test/param_filter_test.go
Comment (staheri14) [inline x/paramfilter/test/param_filter_test.go:42]: > oh true! I can remove the second test case b/c it should be covered by test case one.

In light of this, wouldn't this third (now second) test be covered by the first one as well?

Comment (rootulp) [reply x/paramfilter/test/param_filter_test.go:42]: True. I think we could replace the second test case with a test that verifies that evidence params is actually in the list of `testApp.BlockedParams()...` b/c that's something we don't cover with just the first test case.

I have to modify this PR to account for v1 / v2 branching so I expect that we'll want a standalone test case to verify that evidence params don't exist in v1 and do exist in v2... but will keep this unresolved until then.

**Comment by rootulp** [comment]
Closing as won't do now but we do plan on doing https://github.com/celestiaorg/celestia-app/issues/2418 later.

**Pull request closed without merging**
